test_scripts/pypeit_test.py

```diff
@@ -0,0 +1,1159 @@
+    """Builds a TestSetup object including the tests that it will run"""
+
+    dev_path = os.getenv('PYPEIT_DEV')
+    raw_data = raw_data_dir()
+    outputdir = os.path.abspath(pargs.outputdir)
+
+    # Directory with raw data
+    rawdir = os.path.join(raw_data, instr, setup_name)
+
+    # Directory for reduced data
+    rdxdir = os.path.join(outputdir, instr, setup_name)
+    if not os.path.exists(rdxdir):
+        # Make the directory
+        os.makedirs(rdxdir)
+
+    setup = TestSetup(instr, setup_name, rawdir, rdxdir, dev_path)
+
+    # TODO: By default search for the appropriate pypeit file
+    # and run pypeit_setup if it doesn't exist
+
+    # TODO: Include option that forces the tests to run
+    # pypeit_setup
+    if setup.instr == 'shane_kast_blue' and '600' in setup.name:
+        # Use pypeit_setup to generate the pypeit file
+        setup.tests.append(PypeItSetupTest(setup))
+
+    else:
+        # Use pre-made PypeIt file
+        setup.pyp_file = fix_pypeit_file_directory(setup.dev_path,
+                                                   raw_data_dir(),
+                                                   setup.instr,
+                                                   setup.name,
+                                                   setup.rdxdir)
+
+        # Also try to find and fix pypeit files for
+        # spectroscopic standards.  This is a KLUDGE for
+        # gemini_gmos
+        setup.std_pyp_file = fix_pypeit_file_directory(setup.dev_path,
+                                                       raw_data_dir(),
+                                                       setup.instr,
+                                                       setup.name,
+                                                       setup.rdxdir,
+                                                       std=True)
+
+    # Only want to prep
+    if pargs.prep_only:
+        return setup
+
+    # ----------------------------------------------------------
+    # Reduce tests
+    # ----------------------------------------------------------
+    # Run pypeit
+    if not (flg_after or flg_ql):
+
+        setup.tests.append(PypeItReduceTest(setup, pargs.masters))
+
+        # Run pypeit on any standards, if they exist
+        if setup.std_pyp_file:
+            setup.tests.append(PypeItReduceTest(setup, pargs.masters, std=True))
+
+        # Try re-running keck_lris_red setups with masters
+        if setup.instr == 'keck_lris_red':
+            setup.tests.append(PypeItReduceTest(setup, True))
+
+    if pargs.tests == 'reduce':
+        # Skip the afterburner tests
+        return setup
+    # ----------------------------------------------------------
+
+    if not flg_ql:
+        # ----------------------------------------------------------
+        # SensFunc tests
+        # ----------------------------------------------------------
+        # UVIS algorithm without .sens file
+        if setup.instr == 'shane_kast_blue' and '600' in setup.name:
```

Comment:
I think it would be better to have a list somewhere (up above) 
of instruments that we wish to test fluxing on.  And, if necessary,
also include these `std_file` names.  I'd prefer, though, to have the
name be eg. only `Feige66` and the code would search the `spec1d`
files for any with `Feige66` in it and use the first one.   In case we 
change aspects of the naming convention (possible).